ui/app/components/secret-edit.js

```diff
@@ -197,26 +199,34 @@ export default Component.extend(FocusOnInsertMixin, WithNavToNearestAncestor, {
       secretData.set(secretData.pathAttr, key);
     }
 
-    return secretData.save().then(() => {
-      if (!secretData.isError) {
-        if (isV2) {
-          secret.set('id', key);
+    return secretData
+      .save()
+      .then(() => {
+        if (!secretData.isError) {
+          if (isV2) {
+            secret.set('id', key);
+          }
+          if (isV2 && Object.keys(secret.changedAttributes()).length) {
+            // save secret metadata
+            secret
+              .save()
+              .then(() => {
+                this.saveComplete(successCallback, key);
+              })
+              .catch(e => {
+                this.set(e, e.errors.join(' '));
+              });
+          } else {
+            this.saveComplete(successCallback, key);
+          }
         }
-        if (isV2 && Object.keys(secret.changedAttributes()).length) {
-          // save secret metadata
-          secret
-            .save()
-            .then(() => {
-              this.saveComplete(successCallback, key);
-            })
-            .catch(e => {
-              this.set(e, e.errors.join(' '));
-            });
-        } else {
-          this.saveComplete(successCallback, key);
+      })
+      .catch(error => {
+        if (error instanceof ControlGroupError) {
+          let errorMessage = this.controlGroup.logFromError(error);
+          this.set('error', errorMessage.content);
         }
```

Comment:
To keep the original functionality here, we should `throw error` if it's not an instance of control group error